댓글 추가 API 연동 및 일부 기능 수정

felix-mr opened the pull request:
# 댓글 추가 API 연동 및 일부 기능 수정

## 📎 해당 이슈
#223 

## 🛠 구현 내용 

- 댓글 추가 API 연동
- CardDetail의 content 업데이트 시 네트워크 통신이 성공하면 UI 변경하도록 수정
- CardDetail의 dueDate 업데이트 시 네트워크 통신이 성공하면 UI 변경하도록 수정

## 🙋‍ 리뷰어 참고 사항 
없습니다.

felix-mr committed:
[FEAT] create comment Service / EndPoint 작성

- comment를 추가하는 Service / EndPoint 작성
- comment가 추가된 뒤 화면 갱신

felix-mr committed:
[REFACTOR]  CardDetail 화면 기능 수정

- CardDetail content를 변경하면 네트워크 통신 후 success일 때만 UI 수정
- CardDetail dueData를 변경하면 네트워크 통신 후 success일 때만 UI 수정

felix-mr committed:
[FEAT] 화면 갱신 추가

- member update view controller가 disappear 된 후 card detail 화면 다시 갱신

tksrl0379 commented on iOS/AreUDone/AreUDone/CardDetailScene/View/Controller/CardDetailViewController.swift:
```diff
@@ -387,9 +389,10 @@ extension CardDetailViewController: CardDetailDueDateViewDelegate {
-    DispatchQueue.main.async { [weak self] in
-      self?.stackView.updateDueDateView(with: selectedDate)
+    viewModel.updateDueDate(with: selectedDate) {
+      DispatchQueue.main.async { [weak self] in
+        self?.stackView.updateDueDateView(with: selectedDate)
+      }
```
요런 것도 self?.stackView.updateDueDateView(with: selectedDate) 는 hanlder 로 바인딩해놔서 viewModel 에서 알아서 수행하도록 하면 좋을 것 같은데 어떻게 생각하시나요 ㅎㅎ..

felix-mr replied:
확인했습니다!
바인딩 가능한 부분은 바인딩해서 다시 PR 날리도록 하겠습니다!

felix-mr committed:
[REFACTOR] 리뷰 반영

- 바인딩 할 수 있는 UI를 모두 바인딩
- completionHandler의 형태보다는 binding하여 viewController의 역할을 줄이도록 구현

tksrl0379 reviewed (approved):
수정하느라 수고 많으셨습니다 !!
